### ISSUE TITLE
Vendored ruleset with custom installed_paths does not correctly resolve plugins

### ISSUE THREAD

**Issue by mterwill** [squizlabs/PHP_CodeSniffer#3056]
**Describe the bug**
My organization has several applications that use the same PHP_CodeSniffer standard. Our code standard is a Composer package that is vendored by each application. We commit our vendor/ directory to source control. We use the [PHPCompatibility](https://github.com/PHPCompatibility/PHPCompatibility) plugin in our standard.

**Code sample**
```php
echo "Any file will do.";
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="MyStandard">
    <!-- ... -->
    <rule ref="PHPCompatibility"/>
    <config name="testVersion" value="7.2-7.4"/>
    <config name="installed_paths" value="vendor/phpcompatibility/php-compatibility" />
</ruleset>
```

**To reproduce**
See https://github.com/mterwill/phpcs-vendoring for the full setup. You will need a custom ruleset, vendored into another repository. phpcs error message:

```
ERROR: Referenced sniff "PHPCompatibility" does not exist

Run "phpcs --help" for usage information
```

**Expected behavior**
Anyone should be able to pull down my test repository and run ` ./my-app/script/test` without needing to run another command.

**Versions (please complete the following information):**
 - OS: macOS 10.15.6
 - PHP: 7.3.11
 - PHPCS: 3.5.6

**Additional context**
Something like [DealerDirect/phpcodesniffer-composer-installer](https://github.com/DealerDirect/phpcodesniffer-composer-installer) is not a good fit for anyone who checks in `vendor/`, as you are not required to run `composer install` to use the rest of the app.

**Comment by jrfnl** [comment]
> Something like [DealerDirect/phpcodesniffer-composer-installer][] is not a good fit for anyone who checks in vendor/, as you are not required to run composer install to use the rest of the app.

Help me understand this - you are installing the PHPCompatibility dependency with Composer, are you not ? So the user would have to run `composer install` anyway. So why would the plugin not work for you ? It is the simplest solution for a Composer install-based ruleset.

Heads-up: as of PHPCompatibility 10.0.0, the plugin will automatically install with PHPCompatibility.

**Comment by mterwill** [comment]
My understanding is that DealerDirect/phpcodesniffer-composer-installer configures installed_paths in `CodeSniffer.conf` which is gitignored:

https://github.com/squizlabs/PHP_CodeSniffer/blob/d33a6a99ae2f165f1f835ebf1645e471280cbcbd/.gitignore#L1

Since we version control our `vendor/` directory, there is no need for most engineers to run `composer install`.

**Comment by jrfnl** [comment]
Note: I'm not going to have the discussion about why it's a bad idea to commit the `vendor` directory, but even with that setup, the plugin should work fine.

The plugin will make sure the `CodeSniffer.conf` file in `vendor/squizlabs/php_codesniffer` gets created and filled with the right information on every `install`/`update`. As you say you commit the whole `vendor` directory, that to me means, that you would also commit the `CodeSniffer.conf` file, in which case, everything should work as expected with the plugin in place and individual devs not running `composer install`, but using the dependencies from the committed `vendor` directory.

**Comment by jrfnl** [comment]
@mterwill You can overrule that gitignore (did I already mention it was a bad idea to commit the vendor directory ?)

**Comment by jrfnl** [comment]
> Since we version control our vendor/ directory

You can do that by committing the `composer.lock` file instead, without committing the `vendor` directory.

**Comment by mterwill** [comment]
Force-adding CodeSniffer.conf did the trick for me. Thanks for the quick response @jrfnl!
